pkg/store/bucket.go

```diff
@@ -1472,7 +1483,16 @@ func (r *bucketIndexReader) fetchPostings(groups []*postingGroup) error {
 				r.stats.postingsTouched++
 				r.stats.postingsTouchedSizeSum += len(b)
 
-				_, l, err := r.dec.Postings(b)
+				// Even if this instance is not using compression, there may be compressed
+				// entries in the cache written by other stores.
+				var l index.Postings
+				var err error
+				if isDiffVarintEncodedPostings(b) {
+					l, err = diffVarintDecode(b)
+				} else {
```

Comment:
I would love to avoid `else` everywhere possible... We can do this here as well with duplicated `g.Fill`.. up to you (: It's just whenever I see `else` in Go it's 99% not needed.

Reply:
> It's just whenever I see else in Go it's 99% not needed.

That's a ... interesting... statement :)

Here we would need to duplicate entire
```
				if err != nil {
					return errors.Wrap(err, "decode postings")
				}
				g.Fill(j, l)
```

block. I feels like extra repetition to me, but whatever works...

Reply:
> That's a ... interesting... statement :)

Yes, but I can totally prove this (:

Reply:
Not a blocker here as well, up to you (: